src/Language/DifferentialDatalog/Compile.hs

```diff
@@ -440,15 +457,20 @@ tuple :: [Doc] -> Doc
 tuple [x] = x
 tuple xs = parens $ hsep $ punctuate comma xs
 
-tupleTypeName :: [a] -> Doc
-tupleTypeName xs = "tuple" <> pp (length xs)
+tupleTypeName :: Bool -> [a] -> Doc
+tupleTypeName local xs =
+    tupleTypeName' (if local then "crate" else "::types") xs
```

Comment:
create a function `localPrefix :: bool -> string` for the expression in parenthesis?